### PR TITLE
Problem:(fix #6) HW wallet support not tested

### LINKED ISSUE #6: Problem: HW wallet support not tested
this app may be reused: https://github.com/cosmos/ledger-cosmos
just needs small modifications for the initial experiments (HD-path change, bech32 prefix change...)

Zemu may be used with Golang GRPC for testing: https://github.com/crypto-com/chain/issues/2158#issuecomment-678148494

maybe validator with tmkms could also be tested / tried: https://github.com/tendermint/ledger-validator-app

### PR DESCRIPTION
**Pull request by linfeng-crypto**
solution: add zemu ledger integration test

### CODE REVIEW

File: pystarport/pystarport/cluster.py [hunk 1]
```diff
@@ -635,12 +772,42 @@ def init_cluster(
...
+                acct["address"],
+                account["coins"],
+                vesting_amount=account["coins"],
+                vesting_end_time=vend,
+            )
+
+    ledger = None
+    if with_ledger:
+        ledger = Ledger()
+        ledger.start()
```
[the remark this answers is not in the document]
Comment (linfeng-crypto) [reply pystarport/pystarport/cluster.py]: removed

File: pystarport/pystarport/cluster.py [hunk 2]
```diff
@@ -639,8 +798,33 @@ def init_cluster(
...
     init data directory
     """
+
+    def create_account(cli, account, accounts, use_ledger=False):
+        if use_ledger:
+            acct = cli.create_account_ledger(account["name"])
+        else:
+            acct = cli.create_account(account["name"])
+        print(acct)
+        accounts.append(account)
```
Comment (yihuang) [inline pystarport/pystarport/cluster.py:808]:
```suggestion
        accounts.append(acct)
```

Comment (yihuang) [reply pystarport/pystarport/cluster.py:808]: maybe do the append at outside of this function would be better

File: integration_tests/hardware_wallet/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    cd /tmp && \
+    git clone https://github.com/crypto-com/ledger-rs.git && \
+    cd ledger-rs && \
+    git checkout -b crypto abba8c6cb31dc81b89e24a0132be101432b994b5 && \
+    cd examples/zemu-grpc-server && \
+    cargo build --release
+
+
+FROM zondax/builder-zemu@sha256:4b793ac77c29870e6046e1d0a5019643fd178530205f9cf983bfadd114abca0a
+COPY ./app.elf /home/zondax/speculos/apps/crypto.elf
```
Comment (tomtau) [inline integration_tests/hardware_wallet/Dockerfile:15]: is this built from "ledger-crypto" app?

File: pystarport/pystarport/cluster.py [hunk 4]
```diff
@@ -681,25 +865,17 @@ def init_cluster(
         cli.add_genesis_account(account["address"], node["coins"], i)
         cli.gentx("validator", node["staked"], i)
 
+    # create accounts
     for account in config["accounts"]:
-        acct = cli.create_account(account["name"])
-        print(acct)
+        account = create_account(cli, account)
```
Comment (tomtau) [inline pystarport/pystarport/cluster.py:870]: I guess it should be `acct` or below rename the var to be `account`?

**Review by tomtau — COMMENTED** [review]
runtime error in tests -- I guess it's a wrong variable name used?

File: integration_tests/hardware_wallet/Dockerfile (hunk 3 above)
Comment (DogLi) [reply integration_tests/hardware_wallet/Dockerfile:15]: yes

**Pull request merged**